src/libs/ReactNativeLauncher.js

```diff
@@ -346,6 +346,16 @@ class ReactNativeLauncher extends Launcher {
         job,
         sourceAccountIdentifier
       }
+
+      const hasFilesPermissions = Object.values(
+        manifest.permissions || []
+      ).some(perm => perm.type === 'io.cozy.files')
+      if (hasFilesPermissions) {
+        const existingFilesIndex = await this.getExistingFilesIndex()
+        const serializableExistingFilesIndex =
+          Object.fromEntries(existingFilesIndex)
+        pilotContext.existingFilesIndex = serializableExistingFilesIndex
+      }
```

Comment:
Since the flagship app will have more and more to check the permissions. We should start creating a new domain (aka folder) to deal with that instead of adding the complexity here

Reply:
@Crash-- @acezard I updated the code to have a manifest domain. I think this is ready.

Reply:
Can you add a wrapTime to that method?

Reply:
A wrapTime to `_start` or `hasPermission` ? I'm not sure

Reply:
getExistingFilesIndex ?

Reply:
Already done : https://github.com/cozy/cozy-flagship-app/blob/1a5b00e92f97dc057d65842074a3fbb6c3ff1193/src/libs/ReactNativeLauncher.js#L73